### PR TITLE
HPCC-8614 Improved ConfigMgr

### PR DESCRIPTION
**Pull request by garonsky**

### CODE REVIEW

**Comment by garonsky** [comment]
@rpastrana @xwang2713 please review again
@richardkchapman @stuartort @RussWhitehead FYI

File: deployment/deploy/XMLTags.h [hunk 1]
```diff
@@ -285,7 +328,7 @@
 #define TAG_URL                         "url"
 #define TAG_INSTALLSET                  "installSet"
 #define TAG_PROCESSNAME                 "processName"
-#define TAG_SCHEMA                      "schema"
+#define TAG_SCHEMA                      "SCHEMA "
```
Comment (rpastrana) [inline deployment/deploy/XMLTags.h]: is the white space intentional?

File: configuration/configurator/schemas/SchemaWhiteSpace.hpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
```
Comment (rpastrana) [inline configuration/configurator/schemas/SchemaWhiteSpace.hpp]: Ensure the copyright dates are correct.

File: configuration/configurator/schemas/SchemaUnique.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        pUniqueArray = nullptr;
+    }
+    return pUniqueArray;
+}
+
+void CUniqueArray::dump(::std::ostream &cout, unsigned int offset) const
+{
+    offset+= STANDARD_OFFSET_1;
+
+    quickOutHeader(cout, XSD_UNIQUE_ARRAY_STR, offset);
```
Comment (rpastrana) [inline configuration/configurator/schemas/SchemaUnique.cpp]: Might want to stay consistent with the other functions/macros camel vs all caps+underscores

**Comment by xwang2713** [comment]
Thinks "SCHEMA " should be without whitespace

**Comment by garonsky** [comment]
@RussWhitehead please review

File: cmake_modules/commonSetup.cmake [hunk 4]
```diff
@@ -107,6 +107,13 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
   endif()
   option(LOGGING_SERVICE "Configure use of logging service" ON)
 
+  option(MAKE_CONFIGURATOR "Build Configurator" ON)
+  option(CONFIGURATOR_LIB "Build Configurator static library (.a)" OFF)
+
+  if ( CONFIGURATOR_LIB )
+        set( MAKE_CONFIGURATOR ON )
```
Comment (RussWhitehead) [inline cmake_modules/commonSetup.cmake]: Running configurator with Valgrind reported a lot of memory leaks and asserts. Please investigate

**Review by RussWhitehead — COMMENTED** [review]
Back to you Gleb

**Pull request merged**